src/utils/string-utils.ts

```diff
@@ -1,4 +1,11 @@
-import { isValidRegex } from '.'
+export const isValidRegex = function (str: string): boolean {
```

Comment:
well the file is called string-utils and this is more a general utils or regex utils, so semantically, it does not belong to this file but if it fixes a circular dep, all good from my side

Reply:
true only `isMatchingRegex` is being used directly.
and I am the one to blame apparently https://github.com/PostHog/posthog-js/pull/1698#discussion_r1934416531 lol

Reply:
I'm happy to move them both to `regex-utils`, even though it's only used here. It'll make it more semantic :).